items/src/test/scala/weco/api/items/services/ItemUpdateServiceTest.scala

```diff
@@ -85,59 +90,137 @@ class ItemUpdateServiceTest
     method = AccessMethod.OnlineRequest
   )
 
-  val workWithUnavailableItem = indexedWork().items(
-    List(
-      temporarilyUnavailableItem,
-      dummyDigitalItem
+  class DummyItemUpdater(
+    itemTransform: Seq[Item[IdState.Identified]] => Seq[
+      Item[IdState.Identified]
+    ] = identity
+  ) extends ItemUpdater {
+    override val identifierType: IdentifierType =
+      IdentifierType.SierraSystemNumber
+
+    override def updateItems(
+      items: Seq[Item[IdState.Identified]]
+    ): Future[Seq[Item[IdState.Identified]]] = Future {
+      itemTransform(items)
+    }
+  }
+
+  it("maintains the order of items") {
+    val itemUpdater = new DummyItemUpdater()
+
+    val orderedItems = List(
+      temporarilyUnavailableItem(createSierraItemNumber),
+      createDigitalItem,
+      availableItem(createSierraItemNumber),
+      createDigitalItem,
+      availableItem(createSierraItemNumber),
+      temporarilyUnavailableItem(createSierraItemNumber),
+      createDigitalItem,
+      createDigitalItem
     )
-  )
 
-  val workWithAvailableItem = indexedWork().items(
-    List(
-      availableItem,
-      dummyDigitalItem
+    val reversedItems = orderedItems.reverse
+
+    val workWithItemsForward = indexedWork().items(orderedItems)
+    val workWithItemsBackward = indexedWork().items(reversedItems)
+
+    withItemUpdateService(List(itemUpdater)) { itemUpdateService =>
+      whenReady(itemUpdateService.updateItems(workWithItemsForward)) { items =>
+        items shouldBe orderedItems
+      }
+
+      whenReady(itemUpdateService.updateItems(workWithItemsBackward)) { items =>
+        items shouldBe reversedItems
+      }
+    }
+  }
+
+  it("detects if the item updater returns inconsistent results") {
```

Comment:
What does "inconsistent results" mean? Is this incomplete results (not all items were returned), or something else?

Reply:
Too few or the item updater has returned items with differing IDs. Inconsistent meaning the items updated can't be paired with the items handed for update.

I'll think about how to better convey that.